tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -270,6 +270,10 @@ public TracerSettings(IConfigurationSource source)
             {
                 HttpClientExcludedUrlSubstrings = TrimSplitString(urlSubstringSkips.ToUpperInvariant(), commaSeparator);
             }
+
+            DbmPropagationLevel dbmPropagationValue;
+            Enum.TryParse(source?.GetString(ConfigurationKeys.DbmPropagationMode), true, out dbmPropagationValue);
```

Comment:
Also, any integer here works and returns a valid `DbmPropagationLevel`:
```csharp
Enum.TryParse("100", true, out dbmPropagationValue);
```

because in C# nothing stops you from doing:
```csharp
var value = (DbmPropagationLevel)100; // valid
```

So we should handle that here probably also in `DatabaseMonitoringPropagator.PropagateSpanData()` to be safe.

Reply:
I don't think that the allocation argument is the good one here. We use `Enum.TryParse` in the hot path and here we are in the init phase (even though always better I agree, but we were less picky usually here). 
The fact that we could have another value is actually a strong argument to prevent wrong inputs from the customers (and new values set by RemoteConfig). Thanks for catching it Lucas

Reply:
Yeah, two separate things here 😛
- performance: there's no reason _not_ to use a faster/non-allocating method if one is available, even during initialization
- validation: `Enum.TryParse` is simply not enough for validating the input. I'm not saying "use our enum extensions for validation" because I don't know what our codes does. It's possible our extensions behave the same way as `Enum.TryParse` and also parse any valid integer.

Reply:
I just checked and `[EnumExtensions]` doesn't have anything for parsing, so never mind! We do still need to validate the input, though.

Reply:
It might be easiest to do a string comparison to do this both optimally and concisely. This could be multiple if statements or it could be a switch expression that checks the input using a case-insensitive comparison and assigns the corresponding enum value.